### DISCUSSION TITLE
Slint + .NET C#

### DISCUSSION THREAD

**Discussion by microhobby** [slint-ui/slint#3550] [Show and tell]
I've been working these last few weeks on a binding between Slint UI API and .NET C#, and I'd like to share with you the results. The project is in its initial phase, but it is already possible to create Slint applications with .NET C#, use the C# source generators to create a Window class with the Slint properties and callbacks, and run it on desktop and embedded Linux. Let me know what you think 😎

Repo:
https://github.com/microhobby/slint-dotnet

Video with demos:
https://youtu.be/EwLFhk5RUwE

Blog post: 
https://microhobby.com.br/blog/2023/09/27/creating-user-interface-applications-with-net-and-slint-ui/

Blog post PT-BR:
https://microhobby.com.br/blog/2023/09/27/criando-aplicacoes-com-interface-grafica-com-net-e-slint-ui/

For now it is very experimental, there is work to be done if the community finds it useful...

**Comment by ogoffart** [comment]
I want to start by expressing how genuinely impressed we are with your project! Your work on creating a binding between Slint and .NET C# is truly remarkable, and we are excited about the possibilities it opens up.
Your project aligns perfectly with our vision for Slint, and we believe it can greatly benefit our community. We would be honored if you could consider contributing your work upstream to our repository.

By integrating the C# bindings within our monorepo, they would become first-class citizens. This would enable use of the private API, ensure thorough testing through our CI process, and keep them in sync with the evolution of Slint. 

Ideally, we hope you might be willing to make a pull request to our repository to contribute them. If you wish to continue maintaining it (which we would greatly appreciate), we can grant you write access and ensure proper attribution for your contributions.

**Reply by microhobby** [reply]
I'm glad you liked it, let's work on it! I will open an MR soon.
